src/convertToObject.js

```diff
@@ -12,7 +12,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const callback = (accu, current) => ({
+    ...accu, [current[0]]: current[1],
+  });
+
+  return sourceString
+    .split(';')
+    .map(item => item
+      .trim())
+    .filter(item => item !== '')
+    .map(item => item.split(':'))
+    .map(item => [item[0].trim(), item[1].trim()])
+    .reduce(callback, {});
```

Comment:
**(Optional)** You can simplify your reduce method with Object.fromEntries method. 
Example how it works: 
```

const entries = [
  ['foo', 'bar'],
  ['baz', 42]
]

const obj = Object.fromEntries(entries);

console.log(obj);
// Expected output: Object { foo: "bar", baz: 42 }
```

Reply:
i like it!